src/tree_sitter/array.h

```
@@ -157,82 +186,90 @@ extern "C" {
-          (self->size - index - 1) * element_size);
-  self->size--;
+          (*size - index - 1) * element_size);
+  (*size)--;
 }
 
 /// This is not what you're looking for, see `array_reserve`.
-static inline void _array__reserve(Array *self, size_t element_size, uint32_t new_capacity) {
-  if (new_capacity > self->capacity) {
-    if (self->contents) {
-      self->contents = ts_realloc(self->contents, new_capacity * element_size);
+static inline void *_array__reserve(void *contents, uint32_t *capacity,
```

Clang (14.0.6)

[warning] declaration uses identifier '_array__reserve', which is reserved in the global namespace

(bugprone-reserved-identifier)